torch_xla/distributed/zero_redundancy_optimizer.py

```diff
@@ -324,17 +324,20 @@ def _clip_grad_norm(
         if p.grad is not None:
           p.grad.detach().mul_(clip_value)
 
-  @torch.no_grad()
-  def step(self, closure=None, **kwargs):
-    """
-    Performs a single optimizer step and syncs parameters across all ranks.
-    """
-    assert self.inited, "must call init_zero() first"
-
-    loss = None
-    if closure is not None:
-      with torch.enable_grad():
-        loss = closure()
+  def _get_sharding_scheme(self, **kwargs):
+    if "sharding_scheme" in kwargs:
+    ¦ return kwargs["sharding_scheme"]
+    else:
+    ¦ return [
+    ¦   ¦ {
+    ¦   ¦   ¦ "scale_factor": 1.0,
+    ¦   ¦   ¦ "sharding_group": self.sharding_groups,
+    ¦   ¦   ¦ "group_size": self.local_world_size,
+    ¦   ¦ },
+    ¦ ]
```

Comment:
you seems to copy some formatter metadata in the pr.

Reply:
Thanks, removed